### ISSUE TITLE
RegisterOnMemberUp() does not fire immediately if cluster already up

### ISSUE THREAD

**Issue by ayoung** [akkadotnet/akka.net#2442]
If the cluster member is already up, calling `cluster.RegisterOnMemberUp(callback)` does not immediately call the callback.

According to [this code here](https://github.com/akkadotnet/akka.net/blob/7166307bed5f87e1137518b60d82e57696110bc2/src/core/Akka.Cluster/Cluster.cs#L312)

**Comment by Danthar** [comment]
Hmm. @Horusiath does the JVM work like this as well ?
Im hesitant to mark this as an bug. Because the same issue would apply to all places where you register status callbacks on the cluster extension. Which is due to the async nature of the whole system of course.

I would say this is an problem that needs to be solved in user space. By having the user check the cluster readview for this scenario, should it apply. 
Thoughts? @Aaronontheweb

**Comment by ayoung** [comment]
It seems like an inconsistency with how `RegisterOnMemberRemoved()` works. That method immediately calls the callback if the member is terminated.

**Comment by Aaronontheweb** [comment]
Actually, I wonder if this is true... let me run an experiment real quick.

**Comment by Aaronontheweb** [comment]
@ayoung so it turns out that this isn't an issue, and the test I added in #2446 verifies it.

TL;DR;, the reason for the different code in the `RegisterOnMemberRemoved` is because the actors responsible for executing `RegisterOnMember[X]` callbacks are terminated by that point in time. In the event of `RegisterOnMemberUp`, if a callback is added after the cluster has started the actor will check to see if the current member is already in that state (`MemberUp`), and if so will proceed to invoke the callback immediately rather than wait for another event.
